### PR TITLE
Docblock fixes in TimeInterval

### PR DESCRIPTION
**Pull request by tivnet**
Fixes #
Docblocks only, to please IDEs.
A couple of places `DateTime -> \WC_DateTime`. Others - just `\`

### CODE REVIEW

**Review by adrianduffell — CHANGES_REQUESTED** [review]
@tivnet Thanks for contributing this! Overall it looks good.

 I think ideally we should use [`\DateTimeInterface`](https://www.php.net/manual/en/class.datetimeinterface.php) for all of these return types. It would provide flexibility to change the internal date implementation in the future. Does this work ok in your IDE?

**Comment by tivnet** [comment]
@adrianduffell 

> Does this work ok in your IDE?

In PhpStorm it works fine:

image

Cannot say anything about other IDEs.

**Comment by adrianduffell** [comment]
> In PhpStorm, `@return \DateTimeInterface` works fine:

@tivnet Thanks, let's go ahead with changing it to `\DateTimeInterface` and then I think this will be good to merge 🙂

**Comment by tivnet** [comment]
@adrianduffell Done.

File: src/API/Reports/TimeInterval.php [hunk 1]
```diff
@@ -285,21 +304,23 @@ public static function next_hour_start( $datetime, $reversed = false ) {
 		$hours_offset_timestamp = $timestamp + ( $hour_increment * HOUR_IN_SECONDS - $seconds_into_hour );
 
 		if ( $reversed ) {
-			$hours_offset_timestamp --;
+			$hours_offset_timestamp--;
```
Comment (adrianduffell) [inline src/API/Reports/TimeInterval.php:307]: Let's keep the space before `--`

File: src/API/Reports/TimeInterval.php [hunk 2]
```diff
@@ -30,18 +30,21 @@ class TimeInterval {
 	 * Converts local datetime to GMT/UTC time.
 	 *
 	 * @param string $datetime_string String representation of local datetime.
-	 * @return DateTime
+	 *
```
Comment (adrianduffell) [inline src/API/Reports/TimeInterval.php:33]: A new line was added here.

File: src/API/Reports/TimeInterval.php [hunk 3]
```diff
@@ -247,6 +260,7 @@ public static function intervals_between( $start_datetime, $end_datetime, $inter
 					$start_datetime = self::next_week_start( $start_datetime );
 					$week_count++;
 				} while ( $start_datetime <= $end_datetime );
+
```
Comment (adrianduffell) [inline src/API/Reports/TimeInterval.php:263]: Let's remove the new lines that were added.

File: src/API/Reports/TimeInterval.php [hunk 4]
```diff
@@ -574,10 +608,12 @@ public static function normalize_between_params( $request, $param_names, $is_dat
...
-	 * @param  mixed           $value Parameter value.
-	 * @param  WP_REST_Request $request REST Request.
-	 * @param  string          $param Parameter name.
-	 * @return WP_Error|boolean
+	 * @param mixed            $value   Parameter value.
+	 * @param \WP_REST_Request $request REST Request (Unused).
+	 * @param string           $param   Parameter name.
+	 *
+	 * @return \WP_Error|true
+	 * @noinspection PhpUnusedParameterInspection
```
Comment (adrianduffell) [inline src/API/Reports/TimeInterval.php:616]: Let's remove the @noinspection tag

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Hi @tivnet,

Thanks for making the `\DateTimeInterface` change. However it looks like some other changes came along with it compared to your original commit. I'm guessing it was caused by auto formatting on save. I've highlighted some of it in the code comments.

Are you able to remove these?

Thank you!

**Comment by tivnet** [comment]
@adrianduffell I changed the IDE settings... Now it removed more blanks :)

**Comment by adrianduffell** [comment]
> @adrianduffell I changed the IDE settings... Now it removed more blanks :)

Thanks @tivnet , the removal of the blank lines in the code is a problem too 😅 . Ideally this PR would just be making docblock changes.

**Comment by tivnet** [comment]
@adrianduffell Never mind :)

**Pull request closed without merging**